Client/Impl/Worker/JobWorker.cs

```diff
@@ -77,10 +86,95 @@ internal void Open()
         {
             isRunning = true;
             var cancellationToken = source.Token;
-            var jobWorkerSignal = new JobWorkerSignal();
 
-            StartPollerThread(jobWorkerSignal, cancellationToken);
-            StartHandlerThreads(jobWorkerSignal, cancellationToken);
+            activateJobsCommand = jobWorkerBuilder.Command;
+            maxJobsActive = jobWorkerBuilder.Command.Request.MaxJobsToActivate;
+
+            // Create options for blocks
+            var bufferOptions = new DataflowBlockOptions
+            {
+                CancellationToken = cancellationToken,
+                EnsureOrdered = false
+            };
+
+            var executionOptions = new ExecutionDataflowBlockOptions
+            {
+                MaxDegreeOfParallelism = jobWorkerBuilder.ThreadCount,
+                CancellationToken = cancellationToken,
+                EnsureOrdered = false
+            };
+
+            // Buffer for polled jobs
+            var input = new BufferBlock<IJob>(bufferOptions);
+
+            // Transformblock to process polled jobs
+            var transformer = new TransformBlock<IJob, IJob>(async activatedJob =>
+                {
+                    try
+                    {
+                        await jobHandler(jobClient, activatedJob);
+                        await TryToAutoCompleteJob(activatedJob);
+                    }
+                    catch (Exception exception)
+                    {
+                        await FailActivatedJob(activatedJob, cancellationToken, exception);
+                    }
+                    finally
+                    {
+                        jobClient.Reset();
+                    }
+
+                    return activatedJob;
+                },
+                executionOptions);
+
+            // Action block to finalize handled tasks
+            var output = new ActionBlock<IJob>(activatedJob => { currentJobsActive--; },
+                executionOptions);
+
+            // Link blocks
+            input.LinkTo(transformer);
+            transformer.LinkTo(output);
+
+            // Start polling
+            var pollingTask = Task.Run(async () =>
+                {
+                    while (!source.IsCancellationRequested)
+                    {
+                        if (currentJobsActive >= maxJobsActive)
```

Comment:
In the last release I introduced a threshold to reduce the activation requests, maybe we can bring this back

Reply:
What threshold was it? 60% of MaxJobsToActivate? I think it is not needed because now we are taking into account all jobs activated and still running, so we activate never more than MaxJobsToActivate jobs.

Reply:
The reason for this was more that we send activate commands to early with just one job to activate. If we would wait a bit longer and reach a certain threshold we get a better batching effect. Each activation command is also work for the broker.